cp-main.go

```diff
@@ -282,73 +260,22 @@ func doCopySession(session *sessionV6) {
 	// Enable accounting reader by default.
 	accntReader := newAccounter(session.Header.TotalBytes)
 
-	// Enable progress bar reader only during default mode.
-	var progressReader *barSend
-	if !globalQuiet && !globalJSON { // set up progress bar
-		progressReader = newProgressBar(session.Header.TotalBytes)
-	}
-
 	// Prepare URL scanner from session data file.
-	scanner := bufio.NewScanner(session.NewDataReader())
+	urlScanner := bufio.NewScanner(session.NewDataReader())
 	// isCopied returns true if an object has been already copied
 	// or not. This is useful when we resume from a session.
 	isCopied := isCopiedFactory(session.Header.LastCopied)
 
-	wg := new(sync.WaitGroup)
-	// Limit number of copy routines based on available CPU resources.
-	cpQueue := make(chan bool, int(math.Max(float64(runtime.NumCPU())-1, 1)))
-	defer close(cpQueue)
-
-	// Status channel for receiveing copy return status.
-	statusCh := make(chan copyURLs)
+	// Enable progress bar reader only during default mode.
+	var progressReader *progressBar
+	if !globalQuiet && !globalJSON { // set up progress bar
+		progressReader = newProgressBar(session.Header.TotalBytes)
+	}
 
-	// Go routine to monitor doCopy status and signal traps.
-	wg.Add(1)
+	// Go routine to monitor signal traps.
 	go func() {
-		defer wg.Done()
 		for {
 			select {
```

Comment:
You cannot move this logic outside of select. select prevents race between copy and trap. Otherwise too, if you look at the go routine now, there is no reason for for loop or select and it also leaks.